controllers/factory/k8stools/placeholders.go

```diff
@@ -0,0 +1,34 @@
+package k8stools
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+func RenderPlaceholders[T any](resource *T, placeholders map[string]string) (*T, error) {
+	data, err := json.Marshal(resource)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal resource for filling placeholders: %w", err)
+	}
+
+	if len(placeholders) == 0 || len(data) == 0 {
+		return resource, nil
+	}
+
+	strData := string(data)
+	for placeholderName, value := range placeholders {
+		placeholder := strings.ToUpper(fmt.Sprintf("%%%s%%", placeholderName))
```

Comment:
I don't think that we need such magic. Instead we can validate place holder placeholderName:
- it must starts and ends with %.

Reply:
done